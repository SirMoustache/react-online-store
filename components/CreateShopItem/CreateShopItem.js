/**
 * Absolute imports
 */
import React, { useState } from 'react';
import PropTypes from 'prop-types';

/**
 *
 */
import FileDropZone from '../Forms/FileDropZone';

/**
 * Services
 */
import { uploadImage } from '../../services/fileServices';

const CreateShopItem = ({ onSubmit }) => {
  const [shopItem, setShopItem] = useState({ title: '' });
  const [isUploading, setIsUploading] = useState(false);

  const handleChange = event => {
    const { target } = event;
    const { name, value } = target;
    setShopItem(currentState => ({ ...currentState, [name]: value }));
  };

  const handeSubmit = event => {
    event.preventDefault();
    console.log(shopItem);
    onSubmit({ variables: shopItem });
  };

  const handeFileChange = async acceptedFiles => {
    setIsUploading(true);

    await Promise.all(
      acceptedFiles.map(async file => {
        const uploadResult = await uploadImage(file);
        const preview = uploadResult.secure_url;
        const largePreview = uploadResult.eager[0].secure_url;

        setShopItem(currentState => ({
          ...currentState,
          preview,
          largePreview,
        }));
      }),
    );

    setIsUploading(false);
  };

  return (
    <form onSubmit={handeSubmit}>
      <h4>Create Shop Item</h4>
      <FileDropZone onDrop={handeFileChange} disabled={isUploading} />
      <input
        type="text"
        name="title"
        value={shopItem.title}
        placeholder="Title"
        onChange={handleChange}
      />

      <button type="submit">Create</button>
    </form>
  );
};

CreateShopItem.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default CreateShopItem;
