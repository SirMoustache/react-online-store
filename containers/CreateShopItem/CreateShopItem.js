/**
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';

export const CREATE_SHOP_ITEM_MUTATION = gql`
  mutation CREATE_SHOP_ITEM_MUTATION($title: String!) {
    createShopItem(title: $title) {
      id
    }
  }
`;

const CreateShopItem = ({ render, children }) => {
  const renderer = render || children;

  return (
    <Mutation mutation={CREATE_SHOP_ITEM_MUTATION}>
      {(createShopItem, { error, loading, data }) =>
        renderer({ error, loading, data, createShopItem })
      }
    </Mutation>
  );
};

CreateShopItem.propTypes = {
  render: PropTypes.func,
  children: PropTypes.func,
};

export default CreateShopItem;
