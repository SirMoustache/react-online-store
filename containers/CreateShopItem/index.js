export { default, CREATE_SHOP_ITEM_MUTATION } from './CreateShopItem';
