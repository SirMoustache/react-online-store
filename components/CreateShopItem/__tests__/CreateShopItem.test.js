/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from 'react-testing-library';

/**
 * Local Components
 */
import CreateShopItem from '..';

const renderComponent = ({ ...rest } = {}) =>
  render(<CreateShopItem {...rest} />);

describe('<CreateShopItem/>', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
