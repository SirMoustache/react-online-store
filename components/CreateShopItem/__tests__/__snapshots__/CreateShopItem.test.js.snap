// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateShopItem/> should match snapshot 1`] = `
<div>
  <form>
    <h4>
      Create Shop Item
    </h4>
    <input
      name="title"
      placeholder="Title"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Create
    </button>
  </form>
</div>
`;
